### PR TITLE
Implement the `unflatten` operator

### PR DESCRIPTION
**Pull request by Dakostu**
This change adds an `unflatten` operator that recursively unflattens lists and records according to a heuristic.

### CODE REVIEW

File: libvast/builtins/operators/unflatten.cpp [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+private:
+  std::string separator_ = default_unflatten_separator;
+};
+
+class plugin final : public virtual operator_plugin<unflatten_operator> {
+public:
+  auto parse_operator(parser_interface& p) const -> operator_ptr override {
+    auto parser
+      = argument_parser{"head", "https://vast.io/next/"
```
Comment (dominiklohmann) [inline libvast/builtins/operators/unflatten.cpp:63]:
```suggestion
      = argument_parser{"unflatten", "https://vast.io/next/"
```

File: web/docs/operators/transformations/unflatten.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    "rrtype": "A",
+    "tx_id": 0,
+    "grouped": {
+      "A": [
+        "tenzir.com"
+      ]
+    }
+  }
+}
+
```
Comment (dominiklohmann) [inline web/docs/operators/transformations/unflatten.md:73]:
```suggestion
```

File: web/docs/operators/transformations/unflatten.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+The `unflatten` operator creates nested records out of record entries whose
+names include a separator, thus unflattening
+
+:::info
+`unflatten` uses a heuristic to determine the unflattened schema. Thus, the
+schema of a record that has been flattened using the [`flatten`](flatten.md) operator and
+unflattened afterwards may not be identical to the schema of the unmodified
+record.
+
```
Comment (dominiklohmann) [inline web/docs/operators/transformations/unflatten.md:21]:
```suggestion
```

**Review by dominiklohmann — APPROVED** [review]
This is great, just minor cosmetic feedback.

File: libvast/builtins/operators/unflatten.cpp (hunk 1 above)
Comment (Dakostu) [reply libvast/builtins/operators/unflatten.cpp]: Fixed in https://github.com/tenzir/tenzir/pull/3304/commits/50a4ca3a39b0c885e41cd85e91723a405ae5de82

File: libvast/builtins/operators/unflatten.cpp [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                                "operators/transformations/unflatten"};
+    auto sep = std::optional<located<std::string>>{};
+    parser.add(sep, "<sep>");
```
[the remark this answers is not in the document]
Comment (Dakostu) [reply libvast/builtins/operators/unflatten.cpp]: Fixed in https://github.com/tenzir/tenzir/pull/3304/commits/50a4ca3a39b0c885e41cd85e91723a405ae5de82

File: web/docs/operators/transformations/unflatten.md (hunk 2 above)
Comment (Dakostu) [reply web/docs/operators/transformations/unflatten.md]: Fixed in https://github.com/tenzir/tenzir/pull/3304/commits/50a4ca3a39b0c885e41cd85e91723a405ae5de82

File: web/docs/operators/transformations/unflatten.md (hunk 3 above)
Comment (Dakostu) [reply web/docs/operators/transformations/unflatten.md]: Fixed in https://github.com/tenzir/tenzir/pull/3304/commits/50a4ca3a39b0c885e41cd85e91723a405ae5de82

**Pull request merged**